assets/src/eejs/exceptions/invalid-iso8601-string.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal Imports
+ */
+import InvalidArgument from './invalid-argument';
+
+/**
+ * InvalidIso8601String
+ * Usage: throw new eejs.InvalidISO8601String('some message', [dateTimeString])
+ *
+ * Typically this error is thrown when a given string is not the correct format
+ * for ISO 8601.
+ *
+ * @param {string} msg
+ * @param {mixed} dateTimeString Optional, the timezone string that is invalid
+ */
+export default class InvalidISO8601String extends InvalidArgument {
+	constructor( dateTimeString, message, ...args ) {
+		super( message, dateTimeString, ...args );
+		if ( Error.captureStackTrace ) {
+			Error.captureStackTrace( this, InvalidISO8601String );
+		}
+		this.message = this.message + ' The string provided is not a ' +
+			'valid ISO 8601 formatted string.';
```

Comment:
> How come this class (as well as InvalidLocale and InvalidTimezone below) concatenate the incoming message onto the start of the "base" error message, whereas InvalidDateTime adds the incoming message to the end of the "base" error message.

`InvalidISO8601String`, `InvalidTimeZone`, and `InvalidLocale` all _extend_ `InvalidArgument`.  `InvalidArgument` does this:

```
instance.message = instance.message !== '' ?
		'Invalid argument provided. ' + instance.message :
		'Invalid argument provided.';
```

Which means it takes any provided custom message and appends it to the `InvalidArgument` message and then adds internal message.  So if I passed along a custom message of "Please read up on ISO Strings at http://helpme.com." to `InvalidISO8601String` it'd end up with something like:

> Invalid argument provided. Please read up on ISO Strings at http://helpme.com. The string provided is not a valid ISO 8601 formatted string.

Whereas InvalidDateTime extends TypeError which does not add any custom message (if I understand correctly).

Anyways, although the way you described things is only partially correct, I do agree this can be improved more and I'll do that.